### ISSUE TITLE
Style Guide

### ISSUE THREAD

**Issue by itayyehezkel** [reactjs/he.reactjs.org#22]
Hey guys,

I think we should stay synchronised by adding style guide as @tesseralis mentioned in her [issue](https://github.com/reactjs/he.reactjs.org/issues/1). So I made a few translations which makes sense for me, but feel free to suggest something else. After we all agree about the common language I'll create TRANSLATION.md file.

## Style Guide

| Original  | Translation |
| ------------- | ------------- |
| bind | bind |
| callback | callback  | 
| class | מחלקה |
| cleanup | cleanup |
| component | קומפוננט |
| context | context |
| DOM | DOM |
| feature | תכונה |
| function | פונקציה |
| Hook | Hook |
| key | key |
| library | ספרייה |
| lifecycle | lifecycle |
| props | props |
| React  | React  |
| reconciliation | reconciliation |
| ref | ref |
| state | state |
| TLDR | אמ;לק |
| UI | UI |
| user interface | ממשק משתמש |

**Comment by galnir** [comment]
Yeah it's a great idea
Btw all the translations we pushed so far refer to 'component' as קומפוננטה

**Comment by itayyeh** [comment]
Thanks! Please update the issue because currently I have no permission to change it. I made a [PR](https://github.com/reactjs/reactjs.org-translation/pull/65) to React Translation repository to be a maintainer.

**Comment by itayyehezkel** [comment]
Thanks! please update the issue because currently I have no permission to change it. I made a [PR](https://github.com/reactjs/reactjs.org-translation/pull/65) to be a maintainer.

**Comment by galtalmor** [comment]
Good initiative!
1. Should we actually translate React to ריאקט or leave it in English? I think leaving it in English is more appropriate.
2. Also, I just stumbled upon @yarons' suggestion to translate repository to מאגר, which I disagree with, thinking a clearer translation would be ריפוזיטורי.
What do you guys think of both issues?

**Comment by itayyehezkel** [comment]
Thanks @galtalmor I tend to agree with you, I think that we should listen to our readers, most of Israeli FED industry using English terms and we should not translate every term to Hebrew. And את\ה sounds to me more personal, but I would like to see what others think about it.

**Comment by yarons** [comment]
There are solutions to avoid את/ה/ם, it requires making the text less personal and more generalized so it will approach to women as well.

**Comment by galtalmor** [comment]
@yarons I don't think avoidance is a good solution, it will require the translation to be different from the original, make it harder to translate and as you said will be less personal. I think that choosing either אתה or אתם will be the best solution, we just need to choose one and stick with it. IMO אתם is the better option since it is kind of directed to both sexes, while אתה isn't, but that's only my opinion.

**Comment by yarons** [comment]
@galtalmor 
The translations of Firefox, Wikipedia, Ubuntu, VLC, Mac OS X etc. are all using these rules so it's pretty common to avoid gender (Facebook for example is using dynamic gender customization system so the messages are gender related and usually does not require any additional conjugation).

אתה is the default gender in Hebrew, applies also to women but it's less natural to them so I tend to avoid it.
אתם just doesn't sound right and pretty claustrophobic (who else is with me?).

Do as you wish, I'm just sharing some insights from my personal history, I can only recommend.

**Comment by galnir** [comment]
@yarons I agree with @galtalmor, if we want to avoid את/ה the only option is אתם(which I used in 'thinking in react' translation)

**Comment by yarons** [comment]
@galnir 
As you wish, as I was saying, most well-known open source solutions are translated without gender bias, there's absolutely no reason to use plural form.

**Comment by itayyehezkel** [comment]
What do you think about check with other groups to see how they solved it? maybe all languages with grammatical genders should do the same.

**Comment by yarons** [comment]
@itayyehezkel It's more relevant when you ask yourself whether to translate, transliterate or keep the term as it is, it's less relevant to gender issues, French and Russian for example has a third gender (neutral) so they don't have this problem, In Arabic it's pretty common to use the masculine form (Even love songs that are targeted at women, sang be men, are in masculine form).

Cultural diversity does not really allow having all these comparisons since they don't make any sense.

**Comment by galtalmor** [comment]
Okay, I think we should solve it this way (please vote just once):

/polls "אתה" "את/ה" "אתם" "try to not use any gender"

**Comment by galtalmor** [comment]
Okay lets do it this way:

[![](https://api.gh-polls.com/poll/01D4RFWGAH7C27Z31FC8WSE7NP/%D7%90%D7%AA%D7%94)](https://api.gh-polls.com/poll/01D4RFWGAH7C27Z31FC8WSE7NP/%D7%90%D7%AA%D7%94/vote)
[![](https://api.gh-polls.com/poll/01D4RFWGAH7C27Z31FC8WSE7NP/%D7%90%D7%AA%D7%9D)](https://api.gh-polls.com/poll/01D4RFWGAH7C27Z31FC8WSE7NP/%D7%90%D7%AA%D7%9D/vote)
[![](https://api.gh-polls.com/poll/01D4RFWGAH7C27Z31FC8WSE7NP/avoid%20using%20any%20gender)](https://api.gh-polls.com/poll/01D4RFWGAH7C27Z31FC8WSE7NP/avoid%20using%20any%20gender/vote)

**Comment by dormd** [comment]
production - production

**Comment by dormd** [comment]
What do you think? 
Attribute - תכונה

**Comment by galtalmor** [comment]
I think Attribute should be מאפיין to distinguish between it and feature.
I agree about HTTP header.

**Comment by itayyehezkel** [comment]
Great! Please [join to Slack](https://join.slack.com/t/react-translations/shared_invite/enQtNTQ4MDQxMDQwMTk1LWRjMWI0ZjVlYzUyYmMzN2Q5MmJiNzA1MjdlNjViM2Y0MDQ2ZTRhZmM5NTliYmU1MzRmMTFhNzhkMmU3OTg3MjY) Hebrew channel which can we talk about it in depth.

**Comment by beniregev** [comment]
> Hey guys,
> 
> I think we should stay synchronized by adding a style guide as @tesseralis mentioned in her [issue](https://github.com/reactjs/he.reactjs.org/issues/1). So I made a few translations which makes sense for me, but feel free to suggest something else. After we all agree about the common language I'll create TRANSLATION.md file.
> 
> ## Style Guide
> Original	Translation
> application	אפליקציה
> array	מערך
> attribute	מאפיין
> bind	bind
> callback	callback
> class	מחלקה
> cleanup	נקיון\לנקות
> component	קומפוננטה
> context	context
> constructor	בנאי
> DOM	DOM
> element	אלמנט
> encapsulated	מכומס
> export	ייצוא
> feature	תכונה
> function	פונקציה
> Hook	Hook
> HTTP header	HTTP header
> import	ייבוא
> key	מפתח
> library	ספרייה
> lifecycle	מחזור חיים
> markup	markup
> method	מתודה
> module	מודול
> mount	mount
> node	צומת
> object	אובייקט
> production	production
> props	props
> React	React
> reconciliation	reconciliation
> ref	ref
> render	רנדר\לרנדר
> repository	ריפוזיטורי
> section	חלק
> state	state
> string	מחרוזת
> TLDR	אמ;לק
> UI	ממשק משתמש
> user interface	ממשק משתמש


bind	  קשירה
context	תוכן
constructor	קונסטרקטור
markup	סימון
method	מתודה/שיטה
mount	מאונט
production	סביבת ייצור
props	פרופס
React	ריאקט
reconciliation	הסתרה מחדש
ref	ref
UX חוויית משתמש
user experience חוויית משתמש

**Comment by Sa-ar** [comment]
A few things I stumbled upon in my translation that I'm not sure how to translate:
- Framework
- Deploy
- Serverless
- Adapter (in context of Remix adapter)

**Comment by galnir** [comment]
> A few things I stumbled upon in my translation that I'm not sure how to translate:

In my opinion, these words shouldn't be translated because they are commonly used as-is in the Hebrew speaking tech community

**Comment by beniregev** [comment]
I mainly agree with @galnir , We don't translate "Framework" just write it in Hebrew letters, and the same for "Serverless". However, Adapter is usually translated "מתאם" in Hebrew, and Deploy is "הפצה". I don't know your level of Hebrew, it's not so easy to translate, the translated text into Hebrew is usually shorter than the same text in English. Even between Adapter and Deploy there's a difference - "Adapter" is a noun and "Deploy" a verb - in Hebrew they are treated differently. Are you aware of it?
